fem/bilinearform_ext.cpp

```diff
@@ -558,14 +590,37 @@ void PABilinearFormExtension::Mult(const Vector &x, Vector &y) const
    if (int_face_restrict_lex && iFISz>0)
    {
       int_face_restrict_lex->Mult(x, int_face_X);
-      if (int_face_X.Size()>0)
+      if (int_face_dXdn.Size() > 0)
```

Comment:
With empty local mesh, this is probably not the right check -- every rank needs to call `NormalDerivativeMult` because of the `ExchangeFaceNbrData` call inside.

Reply:
At least this comment seems to imply that: https://github.com/mfem/mfem/blob/4ff5eb8899976a5f389af7a9115b62ffa0974772/fem/prestriction.cpp#L319-L321

Reply:
I think almost all of the PA examples fail with empty local mesh partitions. Perhaps this is something we can address in a separate PR?

(I do think it should be addressed, but debugging/testing will require making some fixes/changes to a lot of other parts of the code).

Reply:
I'm not sure there are that many places where empty partitions are not supported.

I'll try to run a test to see what pops up.

Reply:
I created a branch https://github.com/mfem/mfem/tree/dg-diffusion-empty-parts (based on this branch) which seems to allow `ex14p` to run with empty partitions. I tested with the following change in `ex14p` which introduces one empty partition in the last rank:
```diff
diff --git a/examples/ex14p.cpp b/examples/ex14p.cpp
index 7edb8b0c7..49fe9ae36 100644
--- a/examples/ex14p.cpp
+++ b/examples/ex14p.cpp
@@ -162,7 +162,13 @@ int main(int argc, char *argv[])
    // 5. Define a parallel mesh by a partitioning of the serial mesh. Refine
    //    this mesh further in parallel to increase the resolution. Once the
    //    parallel mesh is defined, the serial mesh can be deleted.
-   ParMesh pmesh(MPI_COMM_WORLD, mesh);
+   int *partitioning = nullptr;
+   {
+      const int np = Mpi::WorldSize();
+      partitioning = mesh.GeneratePartitioning(np > 1 ? np-1 : np);
+   }
+   ParMesh pmesh(MPI_COMM_WORLD, mesh, partitioning);
+   delete [] partitioning;
    mesh.Clear();
    {
       for (int l = 0; l < par_ref_levels; l++)
```
If you want, feel free to merge it here. If not, we can use it as a starting point in a future PR.

Reply:
I'd prefer to do a separate PR (since in my testing it looks like this effects not just DG, but the other examples too) — either you can propose the new PR, or I can do it based on your branch, whatever you prefer.